### PR TITLE
lang: Add support for multiple test suites with separate local validators

### PR DESCRIPTION
**Pull request by paul-schaaf**

### CODE REVIEW

**Comment by paul-schaaf** [comment]
@armaniferrante this is ready for review but a test in the CI is falling and I havent found the reason yet. Im not able to reproduce it locally. if it fails on your machine that might give us a lead

**Comment by paul-schaaf** [comment]
blocked by https://github.com/project-serum/anchor/pull/1653

**Review by tomlinton — COMMENTED** [review]
Tested this out and it worked great. Some minor nits would be renaming `forth-suite` to `fourth-suite` and consistent casing on the ts filenames. No impact on the functionality tough.

File: cli/src/config.rs [hunk 1]
```diff
@@ -531,11 +550,228 @@ fn deser_programs(
...
+                }
+            }
+        }
+        Ok(current_toml)
+    }
+}
+
+fn canonicalize_filepath(s: &mut String, path: impl AsRef<Path>) -> Result<()> {
+    let previous_dir = std::env::current_dir()?;
+    std::env::set_current_dir(path.as_ref().parent().unwrap())?;
```
[the remark this answers is not in the document]
Comment (armaniferrante) [reply cli/src/config.rs:680]: we should add a comment here

**Review by armaniferrante — APPROVED** [review]
some nits. ship it.

**Review by polarisiota — COMMENTED** [review]
this is so useful

**Pull request merged**